sshd-sftp/src/main/java/org/apache/sshd/client/subsystem/sftp/impl/DefaultSftpClient.java

```diff
@@ -501,4 +484,74 @@ public int negotiateVersion(SftpVersionSelector selector) throws IOException {
         versionHolder.set(selected);
         return selected;
     }
+
+    private class SftpChannelSubsystem extends ChannelSubsystem {
```

Comment:
Let's make it protected (and the constructor) in accordance with our policy to allow maximum flexibility to users who wish to modify/enhance the behavior of the relevant class